erpnext/crm/doctype/lead/lead.py

```diff
@@ -104,10 +109,74 @@ def set_lead_name(self):
 
 			self.lead_name = self.company_name
 
+	def create_address(self):
+		address_fields = ["address_title", "address_line1", "address_line2",
+			"city", "county", "state", "country", "pincode"]
+		info_fields = ["email_id", "phone", "fax"]
+
+		# do not create an address if no fields are available,
+		# skipping country since the system auto-sets it from system defaults
+		if not any([self.get(field) for field in address_fields if field != "country"]):
+			return
+
+		address = frappe.new_doc("Address")
+		address.update({addr_field: self.get(addr_field) for addr_field in address_fields})
+		address.update({info_field: self.get(info_field) for info_field in info_fields})
+		address.insert()
+
+		return address
+
+	def create_contact(self):
+		names = self.lead_name.split(" ")
+		if len(names) > 1:
+			first_name, last_name = names[0], " ".join(names[1:])
```

Comment:
This is so ugly and wrong for so many normalization reasons.
Since we are doing changes to lead, could we not just add first_name, last_name on leads to avoid this?

Reply:
The problem is that there's no consistency anywhere, in some places it's first and last name, and contact or user even has a middle name field. Since this is going upstream, I don't want to change other people's implementation with a patch. Even the lead's title is picked from the person name currently. /rant

Reply:
@forellana-digithinkit What we do?

Reply:
@vjFaLk I see @Alchez 's point, its a bigger can of worms to go fix the first/last name thing. This is good enough but not ideal.

Perhaps pushing for normalized fields as a separate PR or a comvo upstream will give us eventual proper normalization.